network_test.go

```diff
@@ -34,6 +36,51 @@ func ExampleNetworkProvider_CreateNetwork() {
 	defer nginxC.Terminate(ctx)
 	nginxC.GetContainerID()
 }
+func Test_NetworkWithIPAM(t *testing.T) {
+	ctx := context.Background()
+	networkName := "test-network-with-ipam"
+	ipamConfig := network.IPAM{
+		Config: []network.IPAMConfig{
+			{
+				Subnet:  "10.1.1.0/24",
+				Gateway: "10.1.1.254/24",
+				IPRange: "10.1.1.1-253",
+			},
+		},
+	}
+	net, _ := GenericNetwork(ctx, GenericNetworkRequest{
+		NetworkRequest: NetworkRequest{
+			Name:           networkName,
+			CheckDuplicate: true,
+			IPAM:           &ipamConfig,
+		},
+	})
+	defer net.Remove(ctx)
+
+	nginxC, _ := GenericContainer(ctx, GenericContainerRequest{
```

Comment:
Would you mind adding a separate test for getting this container's network IPAM from the request? IIUC this test case, the nginx container is not used not asserted.

Reply:
Hey @mdelapenya! Sorry for the delay, but what do you mean by:
> a separate test for getting this container's network IPAM from the request

As I understand now there is already an assertion for it: https://github.com/testcontainers/testcontainers-go/pull/394/files#diff-f84ddb94b392ea0e4f707f6928b01545cbd7faddabe60622c9f348c090fc3e45R82

Reply:
Ahh you are right, I may have missed it 🤦 Thanks for pointing out!